### PR TITLE
[MAINTENANCE] Add default `table_name` to `TableAsset` if omitted

### PR DESCRIPTION
**Pull request by cdkini**
Utilize Pydantic to set `name` as `table_name` if the latter is not provided

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8152&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=8152&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map legend
</details>

File: great_expectations/datasource/fluent/sql_datasource.py [hunk 1]
```diff
@@ -808,6 +808,15 @@ def qualified_name(self) -> str:
             else self.table_name
         )
 
+    @pydantic.validator("table_name", pre=True, always=True)
+    def default_table_name(cls, table_name: str, values: dict, **kwargs) -> str:
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/sql_datasource.py]:
```suggestion
    @pydantic.validator("table_name", pre=True, always=True)
    def _default_table_name(cls, table_name: str, values: dict, **kwargs) -> str:
```
users shouldn't ever need to call this method, so we should make it private.

File: great_expectations/datasource/fluent/sql_datasource.py [hunk 2]
```diff
@@ -808,6 +808,15 @@ def qualified_name(self) -> str:
...
+        if not (validated_table_name := table_name or values.get("name")):
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/sql_datasource.py:816]: 🦭 nice walrus

File: great_expectations/datasource/fluent/sql_datasource.py [hunk 3]
```diff
@@ -797,7 +797,7 @@ class TableAsset(_SQLAsset):
 
     # Instance fields
     type: Literal["table"] = "table"
-    table_name: str
+    table_name: str = ""
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/sql_datasource.py]: `non-blocking`
```suggestion
    table_name: str = Field("", description="Name of the SQL table. Will default to the value of `name` if not provided.")
```
I would use a `pydantic.Field()` here so we can provide some extra metadata to the json schema.
You'll have to run `invoke schema --sync` again after this.

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**